### PR TITLE
Save device name stat prior to running stages

### LINKED ISSUE #80: Benchmarked device name is only recorded at the end of benchmarking
Right now, we save a field to turnkey_stats.yaml called `stats["evaluations"][EVAL_NAME]["device"]` that contains the name of the device under benchmarking (e.g., `Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz`). 

This field is very useful because it shows up in the report CSV and helps to facilitate comparisons across devices.

However, this stat is only saved at the *end* of benchmarking. If the build or benchmark fails, the field gets a value of `-`.

Suggested solution: save the device key to stats as early as possible.

cc @danielholanda @ramkrishna2910

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #80 by making `BaseRT.device_name()` a static method that can be called prior to benchmarking-time. Then, device name is queried before build/benchmark-time and saved as a stat.

The TensorRT runtime is a special case because we need to invoke a docker to learn the device name. In this case, we provide the static `device_name()` method that provides a generic response, and then replace it with a dynamic method and response after the class is instantiated. 

# Versioning

This PR is a small breaking change to the plugin API: the `BaseRT.device_name()` method is now a `@static_method`, and not an `@property`. Downstream plugins will need to be updated accordingly (should be a very small code change). The TKML version number has been revved to `2.0.0` accordingly.

Reviewers: if you can think of a simple way to make this a non-breaking change, let me know. I considered adding a new method like `BaseRT.static_device_name()` in addition to `BaseRT.device_name(self)` but that seemed like a confusing code bloat that wasn't worth it.

# Demos

## Failed build

Here we have a build that fails prior to benchmarking, yet you can see the `device` field is still set correctly. Previously, this field would have been blank.

```
Info: The stats for ~/.cache/turnkey\linear_selftest_76af2f62\turnkey_stats.yaml are:
author: selftest
builtin_model_script: https://github.com/onnx/turnkeyml/tree/cbd3abc173b4bc93d06131f6199d08ef2f6bc143/models/selftest/linear.py
class: LinearTestModel
evaluations:
  57651c20:
    benchmark_status: not_started
    build_status: error
    device: AMD Ryzen 7 PRO 7840U w/ Radeon 780M Graphics
    device_type: x86
```

## TensorRT pre-benchmarking

Prior to benchmarking, TensorRT reports a generic `device`:

```
fdc4a21d:
    benchmark_status: error
    build_status: successful
    device: Nvidia GPU
    device_type: nvidia
```

## TensorRT post-benchmarking

Device is set to a specific value on successful Nvidia benchmarks.

```
fdc4a21d:
    benchmark_status: successful
    build_name: linear_selftest_76af2f62
    build_status: successful
    device: NVIDIA GeForce RTX 4070 Laptop GPU
```

### CODE REVIEW

File: src/turnkeyml/analyze/script.py [hunk 1]
```diff
@@ -667,7 +672,7 @@ def explore_frame(
 
             # Starting in version 2.2.0, torch dynamo added wrappers to callbacks
             # while tracing frames, which conflicts with TurnkeML's analysis. Here,
-            # we supress errors caused by those callback wrappers and only raise an
```
Comment (jeremyfowers) [inline src/turnkeyml/analyze/script.py:670]: fixed a random spelling error

File: src/turnkeyml/analyze/script.py [hunk 2]
```diff
@@ -800,7 +805,7 @@ def forward_spy(*args, **kwargs):
                 and invocation_info.is_target
                 and (model_info.build_model)
             ):
-                # Disable all modifications while we evalute the model
```
Comment (jeremyfowers) [inline src/turnkeyml/analyze/script.py:803]: fixed a random spelling error

File: src/turnkeyml/run/tensorrt/runtime.py [hunk 3]
```diff
@@ -59,18 +59,24 @@ def __init__(
...
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            encoding="utf-8",
-            check=False,
-        )
-
-        if "NVIDIA" not in result.stdout or result.returncode == 1:
-            msg = "No NVIDIA GPUs available on the local machine"
+        msg = "No NVIDIA GPUs available on the local machine"
+        try:
```
Comment (jeremyfowers) [inline src/turnkeyml/run/tensorrt/runtime.py:67]: This was an unrelated bug fix: systems with no Nvidia GPU would crash here when they tried to call nvidia-smi.

File: src/turnkeyml/run/basert.py [hunk 4]
```diff
@@ -260,9 +260,9 @@ def throughput(self) -> float:
         Returns the throughput, in IPS, for the benchmarking run.
         """
 
-    @property
+    @staticmethod
     @abstractmethod
```
Comment (danielholanda) [inline src/turnkeyml/run/basert.py:264]: Does it make sense to have those two decorators in here? Turns out it does!
TIL!
https://stackoverflow.com/questions/4474395/staticmethod-and-abc-abstractmethod-will-it-blend

**Review by danielholanda — APPROVED** [review]
Code is sound. Local execution also works as expected. Approved!
